### PR TITLE
Add ability to ignore lint violations with inline directives

### LINKED ISSUE #65: Add ability to ignore violations
Sometimes it is required to ignore the violations. We need to add a way to do so inline:
- ignore next line
- ignore a block (start/end tokens)

### PR DESCRIPTION
**Pull request by StevenWalker98**
Adds support for inline ignore directives in the lint command:

- `# shellvar-ignore-next-line` — skips the immediately following line
- `# shellvar-ignore-start` / `# shellvar-ignore-end` — skips all lines within the block

Ignored lines are preserved as-is in both lint and fix modes.

### Example

```bash
# This will be reported:
var=$VAR1

# shellvar-ignore-next-line
var=$VAR2  # This is ignored

# shellvar-ignore-start
var=$VAR3  # Ignored
var=$VAR4  # Ignored
# shellvar-ignore-end

var=$VAR5  # This will be reported
```

### Changes

- **`LintCommand::processFile()`** — detects ignore directives and skips flagged lines
- **Fixtures** — `unwrapped-ignore.sh` and `unwrapped-ignore-fixed.sh`
- **Unit test** — `testProcessFileIgnoreDirectives`
- **Functional test** — `testLintCommandFileIgnoreDirectives`

All 422 tests pass.

Closes #65

## Summary

This pull request introduces inline ignore directives for the lint command, enabling developers to selectively suppress linting violations for specific lines or code blocks.

## Changes

**Core Implementation** (`src/Command/LintCommand.php`):
- Modified `processFile()` to detect and handle three ignore directives:
  - `# shellvar-ignore-next-line` — skips the immediately following line
  - `# shellvar-ignore-start` / `# shellvar-ignore-end` — skips all lines within the block
- Tracks ignore state (`ignore_next_line`, `ignore_block`) and preserves ignored lines verbatim without processing via `processLine()`, so ignored content remains unchanged in both lint and fix modes.
- Ensures `ignore_next_line` is correctly reset when entering/exiting ignore blocks to avoid unintended persistence.

**Test Coverage & Fixtures**:
- Unit tests:
  - `testProcessFileIgnoreDirectives()` — validates normal ignore behavior in lint and fix modes.
  - `testProcessFileIgnoreDirectivesEdgeCases()` — covers edge cases (ignore-next-line before block, inside block, consecutive ignore-next-line directives).
  - File: `tests/phpunit/Unit/LintUnitTest.php`
- Functional test:
  - `testLintCommandFileIgnoreDirectives()` — end-to-end lint + fix behavior with directives.
  - File: `tests/phpunit/Functional/LintFunctionalTest.php`
- New fixtures:
  - `tests/phpunit/Fixtures/unwrapped-ignore.sh`
  - `tests/phpunit/Fixtures/unwrapped-ignore-fixed.sh`
  - `tests/phpunit/Fixtures/unwrapped-ignore-edge-cases.sh`

## Testing

All 422 tests pass. New tests confirm:
- Lines and blocks marked with ignore directives are not reported as violations.
- Fix mode applies replacements only to non-ignored violations while preserving ignored lines.
- Fixed output matches expected fixtures.

## Related

Closes #65

### CODE REVIEW

**Comment by AlexSkrypnyk** [comment]
@coderabbitai review

**Comment by AlexSkrypnyk** [comment]
@coderabbitai full review

File: src/Command/LintCommand.php [hunk 1]
```diff
@@ -94,7 +94,37 @@ public function processFile(string $file_name, bool $is_running_fix = FALSE): ar
...
+        $ignore_block = TRUE;
+        $processed_lines[] = $line;
+        continue;
+      }
+
+      if (preg_match('/^#\s*shellvar-ignore-end\s*$/', $trimmed)) {
+        $ignore_block = FALSE;
+        $processed_lines[] = $line;
+        continue;
+      }
```
[the remark this answers is not in the document]
Comment (StevenWalker98) [reply src/Command/LintCommand.php:121]: Fixed in commit b12c203. Added `$ignore_next_line = FALSE;` to both the `shellvar-ignore-start` and `shellvar-ignore-end` handlers to prevent the edge case where a previous `ignore-next-line` directive could cause unintended skipping of lines after block directives.

Tested the fix with the exact reproducer case from the review:
```bash
# shellvar-ignore-next-line
# shellvar-ignore-start
# shellvar-ignore-end
var=$SHOULD_NOT_BE_SKIPPED  # Now correctly reported as violation
```

All 422 tests continue to pass. Thanks @coderabbitai for catching this!

File: src/Command/LintCommand.php [hunk 2]
```diff
@@ -108,12 +108,14 @@ public function processFile(string $file_name, bool $is_running_fix = FALSE): ar
 
       if (preg_match('/^#\s*shellvar-ignore-start\s*$/', $trimmed)) {
         $ignore_block = TRUE;
+        $ignore_next_line = FALSE;
```
Comment (AlexSkrypnyk) [inline src/Command/LintCommand.php:111]: @StevenWalker98 does this requires additional test cases? looks like we are missing on some edge cases if this was not picked up initially

Comment (StevenWalker98) [reply src/Command/LintCommand.php:111]: Good catch — yes, it was missing edge case coverage. Added `testProcessFileIgnoreDirectivesEdgeCases` with a new fixture that covers:

1. `ignore-next-line` followed by a block (`ignore-start`/`ignore-end`) — next-line flag cleared by block start
2. `ignore-next-line` inside a block before `ignore-end` — cleared by block end
3. Consecutive `ignore-next-line` directives — only applies to the next non-directive line

All 423 tests passing (was 422).

Comment (StevenWalker98) [reply src/Command/LintCommand.php:111]: Good catch! You're absolutely right — this edge case should have been covered by tests initially. I'll add a specific test case for the `ignore_next_line` lingering scenario that CodeRabbit identified:

```php
// Test edge case: ignore-next-line before block directive
$content = "# shellvar-ignore-next-line\n# shellvar-ignore-start\n# shellvar-ignore-end\nvar=$SHOULD_BE_REPORTED\n";
```

This will ensure we catch similar edge cases in the future. The current tests focused on the happy path scenarios but missed the interaction between different directive types. Adding this now!
